### ISSUE TITLE
Search patterns with `\V` are case sensitive with smartcase

### ISSUE THREAD

**Issue by infmagic2047** [vim/vim#8705]
**Describe the bug**
Search patterns with `\V` (i.e. switching to very nomagic mode) are considered capital, and therefore case sensitive with smartcase.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Run `vim --clean`
2. Set `ignorecase` and `smartcase`
3. Enter the following text:
```
abc
Abc
ABC
```
3. Search for `\Vabc`, and press `n` to find all matches
4. Only the first line is matched

**Expected behavior**
`\Vabc` is a all-lowercase pattern, so it should match case-insensitively, and therefore matching all three lines.

**Environment (please complete the following information):**
 - Vim version: 8.2.3281
 - OS: Gentoo Linux

**Comment by infmagic2047** [comment]
I believe this is a regression caused by v8.2.3265 (#8682), and the offending code is the following block:
```
else if (*p == '\\' && magic_val == MAGIC_ON)
{
    if (p[1] == '_' && p[2] != NUL)  // skip "\_X"
	p += 3;
    else if (p[1] == '%' && p[2] != NUL)  // skip "\%X"
	p += 3;
    else if (p[1] != NUL)  // skip "\X"
	p += 2;
    else
	p += 1;
}
```

Here, `\X` with alphabetic `X` should be skipped over independent of the magic mode.

**Comment by chrisbra** [comment]
fix should be in #8707

**Comment by tonymec** [comment]
Morality: don't set Very Magic :-P
I set `'smartcase'` in my vimrc, and various ftplugins (including `help.vim` and `html.vim`) use `'ignorecase'` but I always remain at the default Magic value when searching.
What I sometimes do is use a pattern with \C when I want to find *only* lowercase matches.

Best regards,
Tony.
